tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -2513,20 +2514,26 @@ SEC("raw_tracepoint/cgroup_attach_task")
 int tracepoint__cgroup__cgroup_attach_task(struct bpf_raw_tracepoint_args *ctx)
 {
     // Note: we don't place should_trace() here, so we can keep track of the cgroups in the system
+    char cgroup_dirname_buffer[CONT_ID_MIN_FULL_LEN];
     container_id_t container_id = {0};
     struct cgroup *dst_cgrp = (struct cgroup*)ctx->args[0];
     struct task_struct *task = (struct task_struct*)ctx->args[2];
     const char *cgrp_dirname = get_cgroup_dirname(dst_cgrp);
 
-    bpf_probe_read_str(&container_id.id, CONT_ID_LEN+1, cgrp_dirname);
+    // Only update container ID for names longer than 64-bytes.
+    // Container's cgroup dirname should contain 64-bytes of hexadecimal number, so a name smaller than 64-bytes
+    // can't be a real container's name (for example, "docker.service").
+    if (bpf_probe_read_str(&cgroup_dirname_buffer, CONT_ID_MIN_FULL_LEN, cgrp_dirname) == CONT_ID_MIN_FULL_LEN) {
+        bpf_probe_read_str(&container_id.id, CONT_ID_LEN+1, cgrp_dirname);
```

Comment:
Small optional optimization:

Instead of creating a buffer on the stuck, you can use a bigger id field for container_id_t:
```
typedef struct container_id {
    char id[CONT_ID_MIN_FULL_LEN+1];
} container_id_t;
```

Then reading one time instead of twice:
`int cgroup_len = bpf_probe_read_str(&container_id.id, CONT_ID_MIN_FULL_LEN, cgrp_dirname);`
and check if `cgroup_len == CONT_ID_MIN_FULL_LEN`

This should work fine, as we only take CONT_ID_LEN chars in `init_context()` and `sched_process_fork()` events

Reply:
On a second thought, increasing container_id_t will also increase the container id map memory consumption.
We can keep it that way.